backend/src/graphql/resolver/TransactionResolver.test.ts

```diff
@@ -0,0 +1,348 @@
+      // create contribution as user bob
+      const contribution = await mutate({
+        mutation: createContribution,
+        variables: { amount: 1000, memo: 'testing', creationDate: new Date().toISOString() },
+      })
+
+      // login as admin
+      await query({ mutation: login, variables: peterData })
+
+      // confirm the contribution
+      await mutate({
+        mutation: confirmContribution,
+        variables: { id: contribution.data.createContribution.id },
+      })
+
+      // login as bob again
+      await query({ mutation: login, variables: bobData })
+    })
+
+    afterAll(async () => {
+      await cleanDB()
+    })
+
+    describe('trying to send negative amount', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: -50,
+              memo: 'testing negative',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError(`user hasn't enough GDD or amount is < 0`)],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith(
+          `user hasn't enough GDD or amount is < 0 : balance=null`,
+        )
+      })
+    })
+
+    describe('good transaction', () => {
+      it('sends the coins', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 50,
+              memo: 'unrepeateable memo',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            data: {
+              sendCoins: 'true',
+            },
+          }),
+        )
+      })
+
+      it('stores the send transaction event in the database', async () => {
+        // Find the exact transaction (sent one is the one with user[1] as user)
+        const transaction = await Transaction.find({
+          userId: user[1].id,
+          memo: 'unrepeateable memo',
+        })
+
+        expect(EventProtocol.find()).resolves.toContainEqual(
+          expect.objectContaining({
+            type: EventProtocolType.TRANSACTION_SEND,
+            userId: user[1].id,
+            transactionId: transaction[0].id,
+            xUserId: user[0].id,
+          }),
+        )
+      })
+
+      it('stores the receive event in the database', async () => {
+        // Find the exact transaction (received one is the one with user[0] as user)
+        const transaction = await Transaction.find({
+          userId: user[0].id,
+          memo: 'unrepeateable memo',
+        })
+
+        expect(EventProtocol.find()).resolves.toContainEqual(
+          expect.objectContaining({
+            type: EventProtocolType.TRANSACTION_RECEIVE,
+            userId: user[0].id,
+            transactionId: transaction[0].id,
```

Comment:
How can this be the same id? There are two transactions in the DB. One with type receive and one with type send. This looks strange to me.

Reply:
We have two users: `sender` and `receiver`
We create a transaction with that _unique memo_:

- **Sent transaction** - `user: sender` `xUser: receiver` `memo:  'unrepeateable memo'`
-  **Received transaction** - `user: receiver` `xUser: sender` `memo:  'unrepeateable memo'`

Now, if you search by:
`id = sender` && `memo = 'unrepeateable memo'` you will only get the sent transaction, as it is the one assigned to the sender

`id = receiver` && `memo = 'unrepeateable memo'` you will only get the received transaction, as it is the one assigned to the receiver

That is how in both cases `transaction[0]` is used, cause we are finding different transactions in each case